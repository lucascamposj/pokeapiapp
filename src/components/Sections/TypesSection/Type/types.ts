import { ImageSourcePropType } from 'react-native';

interface TypesObjectInterface {
  [key: string]: {
    color: string;
    image: ImageSourcePropType;
  };
}

const typesObject: TypesObjectInterface = {
  bug: {
    color: '#9DC130',
    image: require('../../../../assets/images/pokemonTypes/bug.png'),
  },
  dragon: {
    color: '#0773C7',
    image: require('../../../../assets/images/pokemonTypes/dragon.png'),
  },
  fairy: {
    color: '#EF97E6',
    image: require('../../../../assets/images/pokemonTypes/fairy.png'),
  },
  fire: {
    color: '#F8A54F',
    image: require('../../../../assets/images/pokemonTypes/fire.png'),
  },
  ghost: {
    color: '#6970C5',
    image: require('../../../../assets/images/pokemonTypes/ghost.png'),
  },
  ground: {
    color: '#D78555',
    image: require('../../../../assets/images/pokemonTypes/ground.png'),
  },
  normal: {
    color: '#9A9DA1',
    image: require('../../../../assets/images/pokemonTypes/normal.png'),
  },
  psychic: {
    color: '#F87C7A',
    image: require('../../../../assets/images/pokemonTypes/psychic.png'),
  },
  steel: {
    color: '#5596A4',
    image: require('../../../../assets/images/pokemonTypes/steel.png'),
  },
  shadow: {
    color: '#81A1BE',
    image: require('../../../../assets/images/pokemonTypes/shadow.png'),
  },
  dark: {
    color: '#5F606D',
    image: require('../../../../assets/images/pokemonTypes/dark.png'),
  },
  electric: {
    color: '#EDD53F',
    image: require('../../../../assets/images/pokemonTypes/electric.png'),
  },
  fight: {
    color: '#D94256',
    image: require('../../../../assets/images/pokemonTypes/fight.png'),
  },
  flying: {
    color: '#9BB4E8',
    image: require('../../../../assets/images/pokemonTypes/flying.png'),
  },
  grass: {
    color: '#5DBE62',
    image: require('../../../../assets/images/pokemonTypes/grass.png'),
  },
  ice: {
    color: '#7ED4C9',
    image: require('../../../../assets/images/pokemonTypes/ice.png'),
  },
  poison: {
    color: '#B563CE',
    image: require('../../../../assets/images/pokemonTypes/poison.png'),
  },
  rock: {
    color: '#CEC18C',
    image: require('../../../../assets/images/pokemonTypes/rock.png'),
  },
  water: {
    color: '#559EDF',
    image: require('../../../../assets/images/pokemonTypes/water.png'),
  },
  unknown: {
    color: '#000203',
    image: require('../../../../assets/images/pokemonTypes/unknown.png'),
  },
};

export default typesObject;
