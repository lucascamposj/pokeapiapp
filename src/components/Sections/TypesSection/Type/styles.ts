import styled from 'styled-components/native';

export const Container = styled.View`
  height: 60px;
  width: 50%;
  padding: 0;
  margin: 0;
  justify-content: center;
  align-items: center;
`;

export const TypeImage = styled.Image`
  height: 100%;
  width: 110%;
`;
