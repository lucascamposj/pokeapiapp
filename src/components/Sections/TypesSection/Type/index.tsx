import React from 'react';
import TypesObject from './types';
import { TypeImage, Container } from './styles';
import { TypeInterface } from '../../../../pages/Pokemon';

const Type: React.FC<TypeInterface> = ({ type }: TypeInterface) => {
  const { name } = type;
  const source = TypesObject[name]?.image;
  console.log(name);
  console.log(source);
  return (
    <>
      {source && (
        <Container>
          <TypeImage source={TypesObject[name].image} resizeMode="contain" />
        </Container>
      )}
    </>
  );
};

export default Type;
