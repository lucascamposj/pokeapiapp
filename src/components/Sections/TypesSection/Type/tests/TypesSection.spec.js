import React from 'react';
import {shallow} from 'enzyme';
import Type from '../../Type';

describe('Type', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const type =
              {
                name: "electric",
                url: "electric.com"
              }
            const component = shallow(
              <Type
                type={type}
              />)
            expect(component).toMatchSnapshot()
        });
    });
});
