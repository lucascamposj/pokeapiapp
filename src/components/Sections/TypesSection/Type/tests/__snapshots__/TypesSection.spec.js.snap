// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type Rendering should match to snapshot 1`] = `
<Fragment>
  <Styled(View)>
    <Styled(Image)
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/images/pokemonTypes/electric.png",
        }
      }
    />
  </Styled(View)>
</Fragment>
`;
